pkg/policy/validate.go

```diff
@@ -387,7 +388,7 @@ func ruleForbiddenSectionsHaveVariables(rule *kyverno.Rule) error {
 	var err error
 
 	err = jsonPatchPathHasVariables(rule.Mutation.PatchesJSON6902)
-	if err != nil {
+	if err != nil && errors.Is(errOperationForbidden, err) {
 		return fmt.Errorf("rule \"%s\" should not have variables in patchesJSON6902 path section", rule.Name)
```

Comment:
We only return this error now if it matches our defined error. Previously we recasted ANY error to be this. This caused a lot of confusion when debugging because a specific error was printed when actually something unrelated was wrong.